### PR TITLE
fix: ModularAccountBase cleanup, coverage, custom error types

### PR DESCRIPTION
**Pull request by adamegyed**
## Motivation

Cleanup pass on `ModularAccountBase.sol`.

During this pass, discovered that we lack custom errors for UO and Sig validation module functions.

## Solution

- Add a script to get sizes
- Expand test coverage for MABase
- Remove duplicate error definitions
- Simplify error definitions in MABase
- Add custom errors for all module functions in ExecutionLib.

### CODE REVIEW

File: src/account/ModularAccountBase.sol [hunk 1]
```diff
@@ -87,15 +87,11 @@ abstract contract ModularAccountBase is
...
-    error PreRuntimeValidationHookFailed(address module, uint32 entityId, bytes revertReason);
     error RequireUserOperationContext();
-    error RuntimeValidationFunctionReverted(address module, uint32 entityId, bytes revertReason);
     error SelfCallRecursionDepthExceeded();
-    error SignatureValidationInvalid(address module, uint32 entityId);
-    error UserOpValidationInvalid(address module, uint32 entityId);
-    error UnexpectedAggregator(address module, uint32 entityId, address aggregator);
+    error SignatureValidationInvalid(ModuleEntity validationFunction);
+    error UserOpValidationInvalid(ModuleEntity validationFunction);
+    error UnexpectedAggregator(ModuleEntity validationFunction, address aggregator);
```
Comment (jaypaik) [inline src/account/ModularAccountBase.sol:101]: Why are these`ModuleEntity` while the execution lib errors below are unpacked?

Also nit: Mind sorting this entire block of errors? (see CreateFailed at the bottom)

Comment (adamegyed) [reply src/account/ModularAccountBase.sol:101]: Last I tested, this was both cheaper and smaller in codesize, but let me re-test to be sure.

Comment (adamegyed) [reply src/account/ModularAccountBase.sol:101]: Ok it is actually ever so slightly less codesize - updated!

File: src/libraries/ExecutionLib.sol [hunk 2]
```diff
@@ -196,35 +203,51 @@ library ExecutionLib {
...
                 validationData := mload(0)
             }
+        } else {
+            // Revert with the appropriate error type for the selector used.
+
+            uint32 selectorUsed;
+            uint32 errorSelector;
+
+            assembly ("memory-safe") {
+                selectorUsed := and(mload(add(buffer, 0x4)), 0xffffffff)
```
Comment (Zer0dot) [inline src/libraries/ExecutionLib.sol:241]: clever way to load the first 4 bytes of 0x20!

**Review by jaypaik — APPROVED** [review]

File: src/libraries/ExecutionLib.sol [hunk 3]
```diff
@@ -1029,6 +1057,37 @@ library ExecutionLib {
         return workingMemPtr;
     }
 
+    function _revertModuleFunction(uint32 errorSelector, address moduleAddress, uint32 entityId) private pure {
+        // All of the module function reverts have the same parameter layout:
+        // - module address
+        // - entity Id
+        // - revert data
+
+        assembly ("memory-safe") {
```
Comment (Zer0dot) [inline src/libraries/ExecutionLib.sol:1066]: Is it possible to do 
```
revert(abi.encodeWithselector(this.error.selector, ...)
```

If so, are we sure it'd be cheaper to do this?

**Review by Zer0dot — APPROVED** [review]
small comment, lgtm

File: src/libraries/ExecutionLib.sol (hunk 3 above)
Comment (adamegyed) [reply src/libraries/ExecutionLib.sol:1066]: If we do that, then we'd have to call the high-level function `collectRevertData` first, and the `abi.encode...` expression would copy that memory over to the new memory with the data for the revert. I believe this saves on both gas and codesize due to that added copy.

Comment (Zer0dot) [reply src/libraries/ExecutionLib.sol:1066]: gotcha, lgtm!

**Pull request merged**